modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClusterInfo.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.client.handler;
+
+import java.util.List;
+import java.util.UUID;
+import org.apache.ignite.internal.cluster.management.ClusterTag;
+
+/**
+ * Information about the cluster.
+ */
+public class ClusterInfo {
+    private final ClusterTag tag;
+    private final List<UUID> idHistory;
+
+    /**
+     * Constructor.
+     */
+    public ClusterInfo(ClusterTag tag, List<UUID> idHistory) {
+        this.tag = tag;
+        this.idHistory = List.copyOf(idHistory);
```

Comment:
I don't think we need a defensive copy in the internal API, the caller already creates a new `ArrayList`.

Reply:
Fixed